.github/workflows/tests.yml

```diff
@@ -9,11 +9,96 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-gnu
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/create-build
+        with:
+          target: ${{ matrix.target }}
+      - uses: actions/upload-artifact@v2
+        with:
+          name: flowistry-build
+          path: target/release/${{ matrix.target }}.zip
+
+  test-rust:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+
+    steps:
+      - uses: actions/checkout@v2
+      - run: cargo test
```

Comment:
It's good to use the cargo action instead of the `cargo test` shell command since (IIRC) GitHub can show the output in a more structured way.